### ISSUE TITLE
Support gulpfile.js outside the workspace root

### ISSUE THREAD

**Issue by TDabasinskas** [nickdodd79/vscode-gulptasks#1]
Hi,

It seems that the extensions s[earches for the **gulpfile.js** file only under the workspace root](/nickdodd79/vscode-gulptasks/blob/eb0b3876bf16da72ddf07c2ffb726756baff02e4/src/tasks-loader.ts#L47):

```ts
  const file = path.join(workspaceRoot, 'gulpfile.js');

  if (!await exists(file)) {
    utils.outputInfo('No gulp file found in the current workspace.');
    return emptyTasks;
  }
```

This might be OK in some cases, but there might be cases when the file is nested deeper in the directory. An example case could be a _Visual Studio_ project, with the following structure:
```
─ ProjectFolder
   ├─ .git
   ├─ src
   │    ├─ Program.cs
   │    └─ gulpfile.js
   └─ Project.sln
```

Would it be possible to make it so the extension is able to find a nested **gulpfile.js** as well?

Thank you.

**Comment by nickdodd79** [comment]
Hi @TDabasinskas 

I have some other changes to make so I'll certainly give it a go a part of that.

Nick.

**Comment by nickdodd79** [comment]
Hi @TDabasinskas 

I have published a 0.0.4 version of the extension that now traverses the workspace to discover a gulp file. 

Note that it will use the first one it finds. I am looking to include the ability to discover multiple gulp files with some rework around what the tree displays.

Nick.

**Comment by TDabasinskas** [comment]
Hi @nickdodd79,

It indeed seems to be working just fine now:

```
[Info] Discovering gulp file ...
[Info] Discovered: C:\Projects\Project\src\gulpfile.js
[Info] Loading gulp tasks ...
[Info] Loaded 9 gulp tasks.
```

Thank you!
